### PR TITLE
Feature: Adjust the position of the close [ x ] button of the Dialog component

### PR DESCRIPTION
**Pull request by AnukratiMehta**
## Overview

I changed the position of DialogClose from PrimitiveContent to DialogHeader so that it is properly aligned in the center.

## Changes

- Moved DialogClose to DialogHeader
- Enclosed the div in DialogHeader in a parent div and added DialogClose as the sibling

## Review points

- Your instructions said that "The component structure and styles of DrawerHeader in src/components/ui/Drawer.tsx should be similar," however, I didn't need to make any changes there.  

## Screen Captures

Issue 229

## Assignee Checklist:

- [x] The naming convention of the PR title is correct (See the comment at the top of this template)
- [x] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [x] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [x] The correct assignees and reviewers have been designated for this PR
- [x] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] All the related issues are associated with this PR
- [x] All criteria in the associated issues are met (please tick the checkboxes)
- [x] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [ ] Double-check the "Assignee Checklist"
- [ ] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] The code is well-commented and easy to understand
- [ ] The UI changes accurately reflect the provided design

### CODE REVIEW

File: src/components/ui/Dialog.tsx [hunk 1]
```diff
@@ -63,20 +63,30 @@ const DialogContent = forwardRef<
...
     >
       {children}
-      <DialogClose className={cn('absolute right-3 top-3')}>
+      {/* Move DialogClose to the header */}
+
+      {/* <DialogClose className={cn('')}>
         <Icon icon={CrossIcon} color="black" size={3.5} />
         <span className="sr-only">Close</span>
-      </DialogClose>
+      </DialogClose> */}
```
Comment (nick-y-ito) [inline src/components/ui/Dialog.tsx:71]: Could you please remove the old code and any unnecessary comments related to it? Thank you.

File: src/components/ui/Dialog.tsx [hunk 2]
```diff
@@ -63,20 +63,30 @@ const DialogContent = forwardRef<
...
+    <div
+    className={cn(className)}
     {...props}
   />
+    <DialogClose className={cn('absolute right-3')}>
+        <Icon icon={CrossIcon} color="black" size={3.5} />
+        <span className="sr-only">Close</span>
+      </DialogClose>
+  </div>
+
```
Comment (nick-y-ito) [inline src/components/ui/Dialog.tsx:89]: > Your instructions said that "The component structure and styles of DrawerHeader in src/components/ui/Drawer.tsx should be similar," however, I didn't need to make any changes there.

I apologize for the confusion. What I meant was that the `DialogHeader` should be the same as the `DrawerHeader`, apart from its name. Thus, basically, the only you need to do is to copy and paste from `DrawerHeader` to `DialogHeader.`

**Review by nick-y-ito — CHANGES_REQUESTED** [review]
Thank you for your work! Could you please take a look at the comment I left? Also, there was an error in the GitHub action. Please resolve it.

Here is the error. Do you have an access to view it?
https://github.com/nishiki-tech/nishiki-frontend/actions/runs/7861800958/job/21453642226?pr=234#step:7:1

File: src/components/ui/Dialog.tsx [hunk 3]
```diff
@@ -63,20 +63,30 @@ const DialogContent = forwardRef<
...
+      </DialogClose> */}
     </PrimitiveContent>
   </DialogPortal>
 ));
 DialogContent.displayName = PrimitiveContent.displayName;
 
 const DialogHeader = ({ className, ...props }: HTMLAttributes<HTMLDivElement>) => (
-  <div
-    className={cn('h-12 px-4 border-b border-gray-light', 'flex items-center', className)}
+  // Add a new parent div
```
Comment (nick-y-ito) [inline src/components/ui/Dialog.tsx:78]: You do not need this kind of comment. It is more like a thing that should be added as a PR comment, not a direct comment in the code.

File: src/components/ui/Dialog.tsx (hunk 2 above)
Comment (AnukratiMehta) [reply src/components/ui/Dialog.tsx:89]: Sorry, my bad. It's my first contribution. Just to clarify, I need to undo the changes I made and do this now instead, right?

Comment (nick-y-ito) [reply src/components/ui/Dialog.tsx:89]: No problems! Please practice in our project. We, too, are quite new to accepting contributors.

We don't mind how you commit changes as long as the final code is ok!

**Comment by nick-y-ito** [comment]
@AnukratiMehta 
Additionally, your code does not look formatted correctly. Code formatting is supposed to be performed automatically whenever the `git commit` is executed. If you use a GUI git tool, kindly ensure that the git hook path has been configured correctly.

Moreover, kindly confirm that the file `.husky/_/husky.sh` exists.

**Comment by AnukratiMehta** [comment]
@nick-y-ito 
Thank you so much for the help. The ```.husky/_/husky.sh``` exists and I was able to see "Preparing lint-staged..." when committing changes so everything should work fine now :)

**Comment by nick-y-ito** [comment]
@AnukratiMehta 
Thank you for updating.

 I was wondering if you have access to this [Figma design](https://www.figma.com/file/m0DaNrd0yXWbY4Dy7cuZiN/Nishiki-project?type=design&node-id=0%3A1&mode=design&t=3PHNyHEfFmBxRTdV-1)?

I hope you accurately reflect the design into the code: However, there are still a few remaining issues that need to be addressed:

- [ ] The color of the cross button is not accurately reflected
- [ ] The clickable area of the cross button is too small. Please match its style with the one in the Drawer.
*The entire "DialogHeader" should be very similar to the "DrawerHeader".

Thank you for your attention to these issues.

**Comment by AnukratiMehta** [comment]
@nick-y-ito 

Thanks for your patience. I did, in fact, not realize there was a Figma design. I've addressed the two remaining issues. Hope it works now!

**Review by nick-y-ito — APPROVED** [review]
@AnukratiMehta 
Thank you for the update! I am pleased to inform you that everything looks good and ready to be merged.

**Pull request merged**